### PR TITLE
reuse existing nbm maven plugin to export package

### LINKED ISSUE #4326: Bad version of nbm-maven-plugin on export package generated
### Apache NetBeans version

Apache NetBeans 14

### What happened
ITS VERY IMPORTANT TO FIX THIS, BECOUSE BUILD IS FAILED FROM "THE BOX"

image

When using menuitem "Export package" NB generate :
image

instead of 

image

1. bad version (old as i understand)
2. duplicate plugin definition
3. compile is broken after this operation. This resolving by delete bad plugin declaration and move config to first instance:

image

### How to reproduce

Press "Export package" on any new modul

### Did this work correctly in an earlier version?

Apache NetBeans 12.3 or earlier

### Operating System

Windows 10 21H1

### JDK

17.0.1; Java HotSpot(TM) 64-Bit Server VM 17.0.1+12-LTS-39

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

_No response_

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

### PR DESCRIPTION
**Pull request by ebarboni**
trying to fix #4326

try the different groupid for nbm-maven-plugin to reuse the current one in pom. If none exists create plugin section using the new groupId

### CODE REVIEW

File: apisupport/maven.apisupport/src/org/netbeans/modules/maven/apisupport/MavenPackageModifierImpl.java [hunk 1]
```diff
@@ -211,11 +211,17 @@ private POMExtensibilityElement createPublicPackagesElement(org.netbeans.modules
                             bld = p.getModel().getFactory().createBuild();
                             p.setBuild(bld);
                         }
+                        // check old groupid of nbm maven plugin
                         Plugin plug = bld.findPluginById("org.codehaus.mojo", "nbm-maven-plugin");
+                        // check new groupid of nbm maven plugins
+                        if (plug == null) {
+                            plug = bld.findPluginById("org.apache.netbeans.utilities", "nbm-maven-plugin");
+                        }
```
[the remark this answers is not in the document]
Comment (mbien) [reply apisupport/maven.apisupport/src/org/netbeans/modules/maven/apisupport/MavenPackageModifierImpl.java:219]: the other occurrence is 250-255

[the remark this answers is not in the document]
Comment (ebarboni) [reply apisupport/maven.apisupport/src/org/netbeans/modules/maven/apisupport/MavenPackageModifierImpl.java]: hope is ok (was stressed by the Type but in fact was ok) rebase force pushed

**Review by mbien — APPROVED** [review]
looks good + fix makes sense to me (haven't tested).

NetBeans didn't recognize the old plugin coordinates and simply added the plugin again with the new coordinates which broke the project. Now it won't do that anymore.

File: apisupport/maven.apisupport/src/org/netbeans/modules/maven/apisupport/MavenPackageModifierImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbien) [reply apisupport/maven.apisupport/src/org/netbeans/modules/maven/apisupport/MavenPackageModifierImpl.java]: thanks

**Review by neilcsmith-net — APPROVED** [review]
Looks good.  Should probably consider warnings for old plugins in future?

Will merge once rc1 out and sync completed.

**Comment by mbien** [comment]
> Looks good. Should probably consider warnings for old plugins in future?

yes, had the same thought. There are some maven hints, I never looked into that though.

**Pull request merged**